packages/core/agent/src/launcher/launchctl-daemon.ts

```diff
@@ -0,0 +1,157 @@
+//
+// Copyright 2023 DXOS.org
+//
+
+import { existsSync, mkdirSync } from 'node:fs';
+import { readdir } from 'node:fs/promises';
+import path from 'node:path';
+
+import { waitForCondition } from '@dxos/async';
+import { log } from '@dxos/log';
+
+import { LaunchctlRunner } from './launchctl-runner';
+import { Daemon, ProcessInfo, StartOptions, StopOptions } from '../daemon';
+import { CHECK_INTERVAL, DAEMON_STOP_TIMEOUT } from '../defs';
+import { AgentWaitTimeoutError } from '../errors';
+import { lockFilePath, removeLockFile, removeSocketFile, waitForAgentToStart } from '../util';
+
+/**
+ * Manager of system daemon processes using launchctl on macOS.
+ */
+export class LaunchctlDaemon implements Daemon {
```

Comment:
Why do we separate classes LaunchctlDaemon and LaunchctlRunner? We could refactor LaunchctlDaemon to SystemDaemon, which will be less specific and accept different system-specific launchers.